generated/osdep/meson.build

```diff
@@ -0,0 +1,68 @@
+xcrun = find_program('xcrun')
+swift_prog = find_program(run_command(xcrun, '-find', 'swift').stdout().strip())
+
+# Split the returned output based on '.'
+# The last character of the first array member is the major version.
+# We don't care about potential patch/point releases so we only fetch the
+# minor version.
+ret = run_command(swift_prog, '-version').stdout().split('.')
+swift_ver = ret[0].substring(-1) + '.' + ret[1].substring(0,1)
+
+if swift_ver.version_compare('<4.1')
+    error('A minimum Swift version of 4.1 is required!')
+elif swift_ver.version_compare('>=6.0')
+    error('Swift 6.0 and up is not supported!')
+endif
+
+# custom swift targets
+bridge = join_paths(source_root, 'osdep/macOS_swift_bridge.h')
+header = join_paths(build_root, 'osdep/macOS_swift.h')
+module = join_paths(build_root, 'osdep/macOS_swift.swiftmodule')
+target = join_paths(build_root, 'osdep/macOS_swift.o')
+
+swift_flags = ['-frontend', '-c', '-sdk', macos_sdk_path,
+               '-enable-objc-interop', '-emit-objc-header', '-parse-as-library']
+
+if get_option('debug')
+    swift_flags += '-g'
+endif
+
+if get_option('optimization') != '0'
+    swift_flags += '-O'
+endif
+
+if macos_10_11_features.allowed()
+    swift_flags += ['-D', 'HAVE_MACOS_10_11_FEATURES']
+endif
+
+if macos_10_14_features.allowed()
+    swift_flags += ['-D', 'HAVE_MACOS_10_14_FEATURES']
+endif
+
+extra_flags = get_option('swift-flags').split()
+swift_flags += extra_flags
+
+swift_compile = [swift_prog, swift_flags, '-module-name', 'macOS_swift',
+                 '-emit-module-path', '@OUTPUT0@', '-import-objc-header', bridge,
+                 '-emit-objc-header-path', '@OUTPUT1@', '-o', '@OUTPUT2@',
+                 '@INPUT@', '-I.', '-I' + source_root]
+
+swift_targets = custom_target('swift_targets',
+    input: swift_sources,
+    output: ['macOS_swift.swiftmodule', 'macOS_swift.h', 'macOS_swift.o'],
+    command: swift_compile,
+)
+sources += swift_targets
+
+# go up two directories
+swift_path_split = swift_prog.full_path().split('/')
+remove_dirs = join_paths(swift_path_split.get(-2), swift_path_split.get(-1))
+xcode_dir = swift_prog.full_path().replace(remove_dirs, '')
+swift_lib_dir = join_paths(xcode_dir, 'lib', 'swift', 'macosx')
```

Comment:
i would probably move that to a separate function. the ability to override this with a env var is missing https://github.com/mpv-player/mpv/blob/6265724f3331e3dee8d9ec2b6639def5004a5fa2/waftools/detections/compiler_swift.py#L61-L62

yeah adding the fallback with "xcode-select -p" (https://github.com/mpv-player/mpv/blob/6265724f3331e3dee8d9ec2b6639def5004a5fa2/waftools/detections/compiler_swift.py#L79-L100) would probably be a good idea. who knows what apple will change.